consensus-types/interfaces/beacon_block.go

```diff
@@ -63,30 +57,40 @@ type BeaconBlock interface {
 // that is a beacon block body.
 type BeaconBlockBody interface {
 	RandaoReveal() [field_params.BLSSignatureLength]byte
-	SetRandaoReveal([]byte)
 	Eth1Data() *ethpb.Eth1Data
-	SetEth1Data(*ethpb.Eth1Data)
 	Graffiti() [field_params.RootLength]byte
-	SetGraffiti([]byte)
 	ProposerSlashings() []*ethpb.ProposerSlashing
-	SetProposerSlashings([]*ethpb.ProposerSlashing)
 	AttesterSlashings() []*ethpb.AttesterSlashing
-	SetAttesterSlashings([]*ethpb.AttesterSlashing)
 	Attestations() []*ethpb.Attestation
-	SetAttestations([]*ethpb.Attestation)
 	Deposits() []*ethpb.Deposit
-	SetDeposits([]*ethpb.Deposit)
 	VoluntaryExits() []*ethpb.SignedVoluntaryExit
-	SetVoluntaryExits([]*ethpb.SignedVoluntaryExit)
 	SyncAggregate() (*ethpb.SyncAggregate, error)
-	SetSyncAggregate(*ethpb.SyncAggregate) error
 	IsNil() bool
 	HashTreeRoot() ([field_params.RootLength]byte, error)
 	Proto() (proto.Message, error)
 	Execution() (ExecutionData, error)
-	SetExecution(ExecutionData) error
 	BLSToExecutionChanges() ([]*ethpb.SignedBLSToExecutionChange, error)
+}
+
+type SignedBeaconBlockWriteable interface {
```

Comment:
Seems like SignedBeaconBlock should be a composition of Writable and Readable interfaces no?